pairing/bn256/suite.go

```diff
@@ -16,29 +16,64 @@ import (
 
 // Suite implements the pairing.Suite interface for the BN256 bilinear pairing.
 type Suite struct {
+	*commonSuite
 	g1 *groupG1
 	g2 *groupG2
 	gt *groupGT
-	r  cipher.Stream
 }
 
 // NewSuite generates and returns a new BN256 pairing suite.
 func NewSuite() *Suite {
-	s := &Suite{}
-	s.g1 = &groupG1{}
-	s.g2 = &groupG2{}
-	s.gt = &groupGT{}
+	s := &Suite{commonSuite: &commonSuite{}}
+	s.g1 = &groupG1{commonSuite: s.commonSuite}
+	s.g2 = &groupG2{commonSuite: s.commonSuite}
+	s.gt = &groupGT{commonSuite: s.commonSuite}
 	return s
 }
 
+// SuiteCombined is for convenience, to define a pairing based crypto suite
+// that is at the same time usable as a group.
+type SuiteCombined struct {
```

Comment:
I'd probably make that non-exported if possible.

Reply:
I'm not sure - it might be used in other places outside of bn256.